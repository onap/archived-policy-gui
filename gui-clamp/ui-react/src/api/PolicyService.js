/*-
 * ============LICENSE_START=======================================================
 * ONAP POLICY-CLAMP
 * ================================================================================
 * Copyright (C) 2021 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

export default class PolicyService {
  static getPoliciesList() {
    return fetch(window.location.pathname + 'restservices/clds/v2/policies', {
        method: 'GET',
        credentials: 'same-origin'
        })
        .then(function(response) {
            console.debug("getPoliciesList response received: ", response.status);
            if (response.ok) {
                console.info("getPoliciesList query successful");
                return response.json();
            } else {
                return response.text().then(responseBody => {
                    throw new Error("HTTP " + response.status + "," + responseBody);
                })
            }
        })
        .catch(function(error) {
            console.error("getPoliciesList error occurred ", error);
            alert("getPoliciesList error occurred " + error);
            return undefined;
        })
  }
  static createNewPolicy(policyModelType, policyModelVersion, policyName, policyVersion, policyJson) {
    return fetch(window.location.pathname + 'restservices/clds/v2/policies/' + policyModelType + '/'
                    + policyModelVersion + '/' + policyName + '/' + policyVersion, {
            method: 'POST',
            credentials: 'same-origin',
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify(policyJson)
        })
        .then(function (response) {
            console.debug("createNewPolicy response received: ", response.status);
            if (response.ok) {
                console.info("createNewPolicy query successful");
                return response.text();
            } else {
               return response.text().then(responseBody => {
                    throw new Error("HTTP " + response.status + "," + responseBody);
                })
            }
        })
        .catch(function (error) {
            console.error("createNewPolicy error occurred ", error);
            alert ("createNewPolicy error occurred " + error);
            return undefined;
        });
  }
  static deletePolicy(policyModelType, policyModelVersion, policyName, policyVersion) {
    return fetch(window.location.pathname + 'restservices/clds/v2/policies/' + policyModelType + '/'
        + policyModelVersion + '/' + policyName + '/' + policyVersion, {
            method: 'DELETE',
            credentials: 'same-origin'
        })
        .then(function (response) {
            console.debug("deletePolicy response received: ", response.status);
            if (response.ok) {
                console.info("deletePolicy query successful");
                return response.text();
            } else {
                return response.text().then(responseBody => {
                    throw new Error("HTTP " + response.status + "," + responseBody);
                })
            }
        })
        .catch(function (error) {
            console.error("deletePolicy error occurred ", error);
            alert ("deletePolicy error occurred " + error);
            return undefined;
        });
  }
}
