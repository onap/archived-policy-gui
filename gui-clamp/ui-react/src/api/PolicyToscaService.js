/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019, 2021 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

export default class PolicyToscaService {
  static getToscaPolicyModels() {
    return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels', { method: 'GET', credentials: 'same-origin' })
      .then(function (response) {
        console.debug("getToscaPolicyModels response received: ", response.status);
        if (response.ok) {
          return response.json();
        } else {
          console.error("getToscaPolicyModels query failed");
          return {};
        }
      })
      .catch(function (error) {
        console.error("getToscaPolicyModels error received", error);
        return {};
      });
  }

  static getToscaPolicyModelYaml(policyModelType, policyModelVersion) {
        return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels/yaml/' + policyModelType + "/" + policyModelVersion, {
            method: 'GET',
            credentials: 'same-origin'
        })
            .then(function (response) {
                console.debug("getToscaPolicyModelYaml response received: ", response.status);
                if (response.ok) {
                    return response.json();
                } else {
                    console.error("getToscaPolicyModelYaml query failed");
                    return "";
                }
            })
            .catch(function (error) {
                console.error("getToscaPolicyModelYaml error received", error);
                return "";
            });
 }

 static getToscaPolicyModel(policyModelType, policyModelVersion) {
         return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels/' + policyModelType + "/" + policyModelVersion, {
             method: 'GET',
             credentials: 'same-origin'
         })
             .then(function (response) {
                 console.debug("getToscaPolicyModel response received: ", response.status);
                 if (response.ok) {
                     return response.json();
                 } else {
                     console.error("getToscaPolicyModel query failed");
                     return {};
                 }
             })
             .catch(function (error) {
                 console.error("getToscaPolicyModel error received", error);
                 return {};
             });
  }

  static createPolicyModelFromToscaModel(jsonData) {
       return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels', {
           method: 'POST',
           credentials: 'same-origin',
           headers: {
             "Content-Type": "a",
           },
           body: JSON.stringify(jsonData)
         })
         .then(function(response) {
           console.debug("createPolicyModelFromToscaModel response received: ", response.status);
           if (response.ok) {
             var message = {
               status: response.status,
               message: 'Tosca Policy Model successfully uploaded'
             };
             return message;
           } else {
             console.error("createPolicyModelFromToscaModel failed");
             return response.text();
           }
         })
         .catch(function(error) {
           console.error("createPolicyModelFromToscaModel error received", error);
           return "";
         });
     }

     static updatePolicyModelTosca(policyModelType, policyModelVersion, jsonData) {
         return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels/' + policyModelType + '/' + policyModelVersion, {
             method: 'PUT',
             credentials: 'same-origin',
             headers: {
               "Content-Type": "a",
             },
             body: JSON.stringify(jsonData)
           })
           .then(function(response) {
             console.debug("updatePolicyModelTosca response received: ", response.status);
             if (response.ok) {
               var message = {
                 status: response.status,
                 message: 'Tosca Policy Model successfully uploaded'
               };
               return message;
             } else {
               console.error("updatePolicyModelTosca failed");
               return response.text();
             }
           })
           .catch(function(error) {
             console.error("updatePolicyModelTosca error received", error);
             return "";
           });
       }
}
