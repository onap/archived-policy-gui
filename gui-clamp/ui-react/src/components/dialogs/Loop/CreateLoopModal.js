/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2020 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

import React from 'react'
import Select from 'react-select';
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import Form from 'react-bootstrap/Form';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import styled from 'styled-components';
import LoopService from '../../../api/LoopService';
import TemplateService from '../../../api/TemplateService';
import LoopCache from '../../../api/LoopCache';
import SvgGenerator from '../../loop_viewer/svg/SvgGenerator';

const ModalStyled = styled(Modal)`
	background-color: transparent;
`

const ErrMsgStyled = styled.div`
	color: red;
`

export default class CreateLoopModal extends React.Component {
	constructor(props, context) {
		super(props, context);

		this.getAllLoopTemplates = this.getAllLoopTemplates.bind(this);
		this.handleCreate = this.handleCreate.bind(this);
		this.handleModelName = this.handleModelName.bind(this);
		this.handleClose = this.handleClose.bind(this);
		this.handleDropDownListChange = this.handleDropDownListChange.bind(this);
		this.renderSvg = this.renderSvg.bind(this);
		this.state = {
			show: true,
			chosenTemplateName: '',
			modelInputErrMsg: '',
			modelName: '',
			templateNames: [],
			fakeLoopCacheWithTemplate: new LoopCache({})
		};
	}

	async componentDidMount() {
		await this.getAllLoopTemplates();
		await this.getModelNames();
	}

	handleClose() {
		this.setState({ show: false });
		this.props.history.push('/');
	}

	handleDropDownListChange(e) {
	    if (typeof e.value !== "undefined") {
	        this.setState({
	        fakeLoopCacheWithTemplate:
                new LoopCache({
                    "loopTemplate":e.templateObject,
                    "name": "fakeLoop"
                }),
                chosenTemplateName: e.value
	        })
		} else {
          	this.setState({ fakeLoopCacheWithTemplate: new LoopCache({}) })
        }
	}

	getAllLoopTemplates() {
		TemplateService.getAllLoopTemplates().then(templatesData => {
		    const templateOptions = templatesData.map((templateData) => { return { label: templateData.name, value: templateData.name, templateObject: templateData } });
            this.setState({
                templateNames: templateOptions })
		});
	}

	getModelNames() {
		TemplateService.getLoopNames().then(loopNames => {
			if (!loopNames) {
				loopNames = [];
			}
			// Remove LOOP_ prefix
			let trimmedLoopNames = loopNames.map(str => str.replace('LOOP_', ''));
			this.setState({ modelNames: trimmedLoopNames });
		});
	}

	handleCreate() {
		if (!this.state.modelName) {
			alert("A model name is required");
			return;
		}
		console.debug("Create Model " + this.state.modelName + ", Template " + this.state.chosenTemplateName + " is chosen");
		this.setState({ show: false });
		LoopService.createLoop("LOOP_" + this.state.modelName, this.state.chosenTemplateName).then(text => {
			console.debug("CreateLoop response received: ", text);
			try {
				this.props.history.push('/');
				this.props.loadLoopFunction("LOOP_" + this.state.modelName);
			} catch(err) {
				alert(text);
				this.props.history.push('/');
			}
		})
		.catch(error => {
			console.debug("Create Loop failed");
		});
	}

	handleModelName(event) {
		if (this.state.modelNames.includes(event.target.value)) {
			this.setState({
				modelInputErrMsg: 'A model named "' + event.target.value + '" already exists. Please pick another name.',
				modelName: event.target.value
			});
			return;
		} else {
			this.setState({
				modelInputErrMsg: '',
				modelName: event.target.value
			});
		}
	}

	renderSvg() {
		return (
			<SvgGenerator loopCache={this.state.fakeLoopCacheWithTemplate} clickable={false} generatedFrom={SvgGenerator.GENERATED_FROM_TEMPLATE}/>
		);
	}

	render() {
		return (
			<ModalStyled size="xl" show={this.state.show} onHide={this.handleClose} backdrop="static" keyboard={false} >
				<Modal.Header closeButton>
					<Modal.Title>Create Model</Modal.Title>
				</Modal.Header>
				<Modal.Body>
					<Form.Group as={Row} controlId="formPlaintextEmail">
						<Form.Label column sm="2">Template Name:</Form.Label>
						<Col sm="10">
							<Select onChange={this.handleDropDownListChange} options={this.state.templateNames} />
						</Col>
					</Form.Group>
                    <Form.Group as={Row} style={{alignItems: 'center'}} controlId="formSvgPreview">
                    <Form.Label column sm="2">Model Preview:</Form.Label>
                        <Col sm="10">
                            {this.renderSvg()}
                        </Col>
                    </Form.Group>
					<Form.Group as={Row} controlId="formPlaintextEmail">
						<Form.Label column sm="2">Model Name:</Form.Label>
						<input sm="5" type="text" style={{width: '50%', marginLeft: '1em' }}
							value={this.state.modelName}
							onChange={this.handleModelName}
						/>
						<span sm="5"/>
					</Form.Group>
					<Form.Group as={Row} controlId="formPlaintextEmail">
						<Form.Label column sm="2"> </Form.Label>
						<ErrMsgStyled>{this.state.modelInputErrMsg}</ErrMsgStyled>
					</Form.Group>
				</Modal.Body>
				<Modal.Footer>
					<Button variant="secondary" type="null" onClick={this.handleClose}>Cancel</Button>
					<Button variant="primary" type="submit" onClick={this.handleCreate}>Create</Button>
				</Modal.Footer>
			</ModalStyled>
		);
	}
}
