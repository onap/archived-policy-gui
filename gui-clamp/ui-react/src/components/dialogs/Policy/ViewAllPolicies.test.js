/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2022 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import {shallow} from "enzyme";
import React from "react";
import ViewAllPolicies from "./ViewAllPolicies";
import fs from "fs";
import PolicyToscaService from "../../../api/PolicyToscaService";
import PolicyService from "../../../api/PolicyService";
import CreateLoopModal from "../Loop/CreateLoopModal";
import toJson from "enzyme-to-json";

describe('Verify ViewAllPolicies', () => {
    let toscaPolicyModels = fs.readFileSync('src/components/dialogs/Policy/toscaData.test.json', {
        encoding: 'utf8',
        flag: 'r'
    });
    let toscaPoliciesList = fs.readFileSync('src/components/dialogs/Policy/toscaPoliciesList.test.json', {
        encoding: 'utf8',
        flag: 'r'
    });

    it("renders correctly", () => {
        const component = shallow(<ViewAllPolicies />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('Test handleClose', () => {
        const historyMock = {push: jest.fn()};
        const handleClose = jest.spyOn(ViewAllPolicies.prototype, 'handleClose');
        const component = shallow(<ViewAllPolicies history={historyMock}/>);

        component.find('[variant="secondary"]').prop('onClick')();

        expect(handleClose).toHaveBeenCalledTimes(1);
        expect(component.state('show')).toEqual(false);
        expect(historyMock.push.mock.calls[0]).toEqual(['/']);

        handleClose.mockClear();
    });

    it('handleClose called when top-right button clicked', () => {
        const historyMock = {push: jest.fn()};
        const handleClose = jest.spyOn(ViewAllPolicies.prototype, 'handleClose');
        const component = shallow(<ViewAllPolicies history={historyMock}/>);

        component.find('[size="xl"]').get(0).props.onHide();

        expect(handleClose).toHaveBeenCalledTimes(1);
        expect(component.state('show')).toEqual(false);
        expect(historyMock.push.mock.calls[0]).toEqual(['/']);

        handleClose.mockClear();
    });

    it('Test getAllToscaModels', () => {
        PolicyToscaService.getToscaPolicyModels = jest.fn().mockImplementation(() => {
            return Promise.resolve(toscaPolicyModels);
        });

        const getAllToscaModels = jest.spyOn(ViewAllPolicies.prototype, 'getAllToscaModels');
        shallow(<ViewAllPolicies/>);

        expect(getAllToscaModels).toHaveBeenCalledTimes(1);
    });

    it('Test setToscaModelsListData', async () => {
        PolicyToscaService.getToscaPolicyModel = jest.fn().mockImplementation(() => {
            return Promise.resolve(toscaPolicyModels);
        });

        const component = shallow(<ViewAllPolicies/>);
        component.setState({toscaModelsListData: toscaPolicyModels})
        expect(component.state('toscaModelsListData')).toEqual(toscaPolicyModels);
    });

    it('Test getAllPolicies', () => {
        PolicyService.getPoliciesList = jest.fn().mockImplementation(() => {
            return Promise.resolve(toscaPoliciesList);
        });

        const getAllPolicies = jest.spyOn(ViewAllPolicies.prototype, 'getAllPolicies');
        shallow(<ViewAllPolicies/>);

        expect(getAllPolicies).toHaveBeenCalledTimes(1);
    });

    it('Test setPoliciesListData', async () => {
        PolicyService.getPoliciesList = jest.fn().mockImplementation(() => {
            return Promise.resolve(toscaPoliciesList);
        });

        const component = shallow(<ViewAllPolicies/>);
        component.setState({policiesListData: toscaPoliciesList});
        component.setState({policiesListDataFiltered: toscaPoliciesList});

        expect(component.state('policiesListData')).toEqual(toscaPoliciesList);
        expect(component.state('policiesListDataFiltered')).toEqual(toscaPoliciesList);
    });

    it('Test handleDeletePolicy event rowdata', async () => {
        const rowData = {
            type: 'onap.policies.monitoring.tcagen2',
            type_version: '1.0.0',
            name: 'MICROSERVICE_vLoadBalancerMS_v1_0_tcagen2_1_0_0_AV0',
            version: '1.0.0'
        }

        PolicyService.deletePolicy = await jest.fn().mockImplementation(() => {
            return Promise.resolve({});
        });

        const event = { target: {value: 'event'}}
        const component = shallow(<CreateLoopModal/>);

        component.setState({showSuccessAlert: true});
        component.setState({showMessage: 'Policy successfully Deleted'});

        component.find('input').simulate('click', event, rowData);
        component.update();

        expect(component.state('showSuccessAlert')).toEqual(true);
        expect(component.state('showMessage')).toEqual('Policy successfully Deleted');
    });
});
