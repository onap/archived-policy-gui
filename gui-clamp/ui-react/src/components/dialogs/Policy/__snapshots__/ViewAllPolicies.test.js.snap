// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify ViewAllPolicies renders correctly 1`] = `
<Fragment>
  <Styled(Modal)
    backdrop="static"
    keyboard={false}
    onHide={[Function]}
    show={true}
    size="xl"
  >
    <ModalHeader
      closeButton={true}
      closeLabel="Close"
    />
    <Tabs
      id="controlled-tab-example"
      mountOnEnter={false}
      onSelect={[Function]}
      unmountOnExit={false}
      variant="tabs"
    >
      <Tab
        eventKey="policies"
        title="Policies in Policy Framework"
      >
        <ModalBody>
          <div>
            <styled.div>
              <PoliciesTreeViewer
                policiesData={Array []}
                policiesFilterFunction={[Function]}
                valueForTreeCreation="name"
              />
            </styled.div>
            <styled.div>
              <WithStyles(Component)
                actions={
                  Array [
                    Object {
                      "icon": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      "onClick": [Function],
                      "tooltip": "Delete Policy",
                    },
                  ]
                }
                columns={
                  Array [
                    Object {
                      "cellStyle": Object {
                        "backgroundColor": "#039be5",
                        "border": "1px solid black",
                        "color": "#FFF",
                      },
                      "field": "name",
                      "headerStyle": Object {
                        "backgroundColor": "#ddd",
                        "border": "2px solid black",
                      },
                      "title": "Policy Name",
                    },
                    Object {
                      "cellStyle": Object {
                        "backgroundColor": "#039be5",
                        "border": "1px solid black",
                        "color": "#FFF",
                      },
                      "field": "version",
                      "headerStyle": Object {
                        "backgroundColor": "#ddd",
                        "border": "2px solid black",
                      },
                      "title": "Policy Version",
                    },
                    Object {
                      "cellStyle": Object {
                        "backgroundColor": "#039be5",
                        "border": "1px solid black",
                        "color": "#FFF",
                      },
                      "field": "type",
                      "headerStyle": Object {
                        "backgroundColor": "#ddd",
                        "border": "2px solid black",
                      },
                      "title": "Policy Type",
                    },
                    Object {
                      "cellStyle": Object {
                        "backgroundColor": "#039be5",
                        "border": "1px solid black",
                        "color": "#FFF",
                      },
                      "field": "type_version",
                      "headerStyle": Object {
                        "backgroundColor": "#ddd",
                        "border": "2px solid black",
                      },
                      "title": "Policy Type Version",
                    },
                    Object {
                      "cellStyle": Object {
                        "backgroundColor": "#039be5",
                        "border": "1px solid black",
                        "color": "#FFF",
                      },
                      "field": "supportedPdpGroupsString",
                      "headerStyle": Object {
                        "backgroundColor": "#ddd",
                        "border": "2px solid black",
                      },
                      "title": "Deployable in PDP Group",
                    },
                    Object {
                      "cellStyle": Object {
                        "backgroundColor": "#039be5",
                        "border": "1px solid black",
                        "color": "#FFF",
                      },
                      "field": "pdpGroupInfoString",
                      "headerStyle": Object {
                        "backgroundColor": "#ddd",
                        "border": "2px solid black",
                      },
                      "title": "Deployed in PDP Group",
                    },
                  ]
                }
                data={Array []}
                detailPanel={
                  Array [
                    Object {
                      "icon": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      "render": [Function],
                      "tooltip": "Show Configuration",
                    },
                    Object {
                      "icon": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      "openIcon": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      "render": [Function],
                      "tooltip": "Show Raw Data",
                    },
                    Object {
                      "icon": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      "openIcon": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      "render": [Function],
                      "tooltip": "PDP Group Deployment",
                    },
                  ]
                }
                icons={
                  Object {
                    "Add": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Check": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Clear": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Delete": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "DetailPanel": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Edit": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Export": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Filter": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "FirstPage": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "LastPage": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "NextPage": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "PreviousPage": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "ResetSearch": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Search": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "SortArrow": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "ThirdStateCheck": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "ViewColumn": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  }
                }
                onRowClick={[Function]}
                options={
                  Object {
                    "actionsColumnIndex": -1,
                    "exportButton": true,
                    "grouping": true,
                    "headerStyle": Object {
                      "backgroundColor": "#ddd",
                      "border": "1px solid black",
                      "fontSize": "15pt",
                      "text": "bold",
                    },
                  }
                }
                title="Policies"
              />
            </styled.div>
          </div>
        </ModalBody>
      </Tab>
      <Tab
        eventKey="tosca models"
        title="Tosca Models in Policy Framework"
      >
        <ModalBody>
          <div>
            <styled.div>
              <PoliciesTreeViewer
                policiesData={Array []}
                policiesFilterFunction={[Function]}
                valueForTreeCreation="policyModelType"
              />
            </styled.div>
            <styled.div>
              <WithStyles(Component)
                actions={
                  Array [
                    Object {
                      "icon": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      "isFreeAction": true,
                      "onClick": [Function],
                      "tooltip": "Add New Tosca Model",
                    },
                  ]
                }
                columns={
                  Array [
                    Object {
                      "cellStyle": Object {
                        "backgroundColor": "#039be5",
                        "border": "1px solid black",
                        "color": "#FFF",
                      },
                      "field": "policyModelType",
                      "headerStyle": Object {
                        "backgroundColor": "#ddd",
                        "border": "2px solid black",
                      },
                      "title": "Policy Model Type",
                    },
                    Object {
                      "cellStyle": Object {
                        "backgroundColor": "#039be5",
                        "border": "1px solid black",
                        "color": "#FFF",
                      },
                      "field": "policyAcronym",
                      "headerStyle": Object {
                        "backgroundColor": "#ddd",
                        "border": "2px solid black",
                      },
                      "title": "Policy Acronym",
                    },
                    Object {
                      "cellStyle": Object {
                        "backgroundColor": "#039be5",
                        "border": "1px solid black",
                        "color": "#FFF",
                      },
                      "field": "version",
                      "headerStyle": Object {
                        "backgroundColor": "#ddd",
                        "border": "2px solid black",
                      },
                      "title": "Version",
                    },
                    Object {
                      "cellStyle": Object {
                        "backgroundColor": "#039be5",
                        "border": "1px solid black",
                        "color": "#FFF",
                      },
                      "field": "updatedBy",
                      "headerStyle": Object {
                        "backgroundColor": "#ddd",
                        "border": "2px solid black",
                      },
                      "title": "Uploaded By",
                    },
                    Object {
                      "cellStyle": Object {
                        "backgroundColor": "#039be5",
                        "border": "1px solid black",
                        "color": "#FFF",
                      },
                      "editable": "never",
                      "field": "updatedDate",
                      "headerStyle": Object {
                        "backgroundColor": "#ddd",
                        "border": "2px solid black",
                      },
                      "title": "Uploaded Date",
                    },
                  ]
                }
                data={Array []}
                detailPanel={
                  Array [
                    Object {
                      "icon": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      "render": [Function],
                      "tooltip": "Show Tosca",
                    },
                    Object {
                      "icon": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      "openIcon": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      "render": [Function],
                      "tooltip": "Show Raw Data",
                    },
                    Object {
                      "icon": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      "openIcon": Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      "render": [Function],
                      "tooltip": "Create a policy from this model",
                    },
                  ]
                }
                icons={
                  Object {
                    "Add": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Check": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Clear": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Delete": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "DetailPanel": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Edit": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Export": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Filter": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "FirstPage": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "LastPage": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "NextPage": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "PreviousPage": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "ResetSearch": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "Search": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "SortArrow": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "ThirdStateCheck": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "ViewColumn": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  }
                }
                onRowClick={[Function]}
                options={
                  Object {
                    "actionsColumnIndex": -1,
                    "exportButton": true,
                    "grouping": true,
                    "headerStyle": Object {
                      "backgroundColor": "#ddd",
                      "border": "1px solid black",
                      "fontSize": "15pt",
                      "text": "bold",
                    },
                  }
                }
                title="Tosca Models"
              />
            </styled.div>
          </div>
        </ModalBody>
      </Tab>
    </Tabs>
    <Alert
      closeLabel="Close alert"
      dismissible={true}
      onClose={[Function]}
      show={false}
      transition={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "appear": false,
            "in": false,
            "mountOnEnter": false,
            "timeout": 300,
            "unmountOnExit": false,
          },
          "render": [Function],
        }
      }
      variant="success"
    >
      <styled.div />
    </Alert>
    <Alert
      closeLabel="Close alert"
      dismissible={true}
      onClose={[Function]}
      show={false}
      transition={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "appear": false,
            "in": false,
            "mountOnEnter": false,
            "timeout": 300,
            "unmountOnExit": false,
          },
          "render": [Function],
        }
      }
      variant="danger"
    >
      <styled.div />
    </Alert>
    <ModalFooter>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        variant="secondary"
      >
        Close
      </Button>
    </ModalFooter>
  </Styled(Modal)>
  <PolicyToscaFileSelector
    disableFunction={[Function]}
    show={false}
    toscaTableUpdateFunction={[Function]}
  />
</Fragment>
`;
